iopipe/tests/test_agent.py

```diff
@@ -83,52 +83,57 @@ def test_erroring(handler_that_errors, mock_context):
 # Advanced reporting
 @pytest.fixture
 def advanced_handler(iopipe):
-    mock_context = MockContext('advancedHandler', '1')
+    nested_iopipe = []
 
     @iopipe.decorator
     def _advanced_handler(event, context):
-        nested_iopipe = IOpipe('test-suite')
+        nested_iopipe.append(IOpipe('test-suite'))
         timestamp = time.time()
-        report = nested_iopipe.create_report(timestamp, context)
+        report = nested_iopipe[0].create_report(timestamp, context)
         try:
             pass
         except Exception as e:
-            nested_iopipe.err(e)
-        # FIXME: For some reason the finally block isn't called for a nested
+            nested_iopipe[0].err(e)
         report.update_data(context, timestamp)
         report.send()
-        return nested_iopipe
 
-    return _advanced_handler(None, mock_context)
+    return nested_iopipe, _advanced_handler
```

Comment:
No, it is an array because of the way Python scopes nested functions. Python 3 now has a `nonlocal` keyword to mutate variables defined in the wrapping function from within the nested function, but Python 2 doesn't support this. So the workaround is to use a list.

For example:

```python
def foo():
  bar = False
  def baz():
    bar = True 
  baz()
  return bar
```

Will return `False`, but if `bar` was a list and `baz` just appended to it then `bar` can be mutated from within `baz`.